### ISSUE TITLE
Quotient module test failure

### ISSUE THREAD

**Issue by wbhart** [Nemocas/AbstractAlgebra.jl#430]
The current QuotientModule-test.jl fails at line 52. Here is a more or less minimal example:

```
         F = FreeModule(ZZ, 3)
         m = F(BigInt[0, 0, 10])
         n = F(BigInt[4, 6, 2])
         S, f1 = sub(F, [m, n])
         M, g1 = quo(F, S)
         S = [M(BigInt[0, -5, 7])]
         N, f = sub(M, S)
         T = [N(BigInt[0]), N(BigInt[-1])]
         P, g = sub(N, T)
         Q1, h1 = quo(M, P)

         U = [f(v) for v in T]
         V, k = sub(M, U)

         Q2, h2 = quo(M, V)

         @test Q1 == Q2
```

It only seems to fail over ZZ, not QQ and it only seems to fail when there is precisely 1 element in S and at least 2 in T. I don't find any failures with a free module F of rank 2.

**Comment by wbhart** [comment]
The current test code only produces a failure about every 50,000 iterations which is why we haven't seen this before.

**Comment by wbhart** [comment]
Q1 and Q2 look the same to me. The question is why don't they compare equal. Their relations don't seem to be canonicalised somehow.

**Comment by thofma** [comment]
I am afraid you are on your own with this one :(

**Comment by wbhart** [comment]
Seems to be (another) bug in HNF (we really ought to test that code super thoroughly some day), unless I am missing something obvious (which I probably am).

See #438

**Comment by thofma** [comment]
Fixed by #444
